src/main/java/javaslang/collection/Traversable.java

```diff
@@ -553,6 +538,73 @@ default T fold(T zero, BiFunction<? super T, ? super T, ? extends T> op) {
     Option<T> headOption();
 
     /**
+     * Returns the index of the first occurrence of the given element or -1 if this does not contain the given element.
+     *
+     * @param element an element
+     * @return the index of the first occurrence of the given element
+     */
+    default int indexOf(T element) {
+        return indexOf(element, 0);
+    }
+
+    /**
+     * Returns the index of the first occurrence of the given element after or at some start index
+     * or -1 if this does not contain the given element.
+     *
+     * @param element an element
+     * @param from start index
+     * @return the index of the first occurrence of the given element
+     */
+    int indexOf(T element, int from);
+
+    /**
+     * Finds first index where this sequence contains a given sequence as a slice.
+     * <p>
+     * Note: may not terminate for infinite-sized collections.
+     * @param that the sequence to test
+     * @return the first index such that the elements of this sequence starting at this index match
+     * the elements of sequence that, or -1 of no such subsequence exists.
+     */
+    default int indexOfSlice(Iterable<? extends T> that) {
+        return indexOfSlice(that, 0);
```

Comment:
Also here I would insert `Objects.requireNonNull(that, "that is null");`

and also declare a javadoc param `@throws NullPointerException if that is null`.